### PR TITLE
[Refactor] refactor requirements 

### PR DESCRIPTION
**Pull request by teamwong111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Speed up compilation. According to my test, compilation with ninja is more than three times faster than compilation with setuptools.build_ext. Additionally, using ninja is recommended by PyTorch Offical.

## Modification

Modify requirements.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
`ninjia` is not a required package if users do not build from source. Maybe it can be added to optional.txt.

**Comment by zhouzaida** [comment]
We can refer to https://github.com/open-mmlab/mmflow/pull/11 to refactor the requirements of mmcv.

**Review by zhouzaida — APPROVED** [review]

**Comment by ZwwWayne** [comment]
need to resolve conflicts before merging

**Review by zhouzaida — APPROVED** [review]

**Comment by teamwong111** [comment]
> need to resolve conflicts before merging

done

File: requirements/optional.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+ninja
```
Comment (ZwwWayne) [inline requirements/optional.txt:1]: If pytorch also installs ninja, shall we make it a default requirements to speed up compilation?

Comment (zhouzaida) [reply requirements/optional.txt:1]: Most users will use pre-compiled packages, so this library is not required. Also, if you use `conda` to install PyTorch, then ninja is automatically installed.

**Pull request merged**